service/dap/server.go

```diff
@@ -2174,8 +2174,10 @@ func (s *Session) maybeLoadResliced(v *fullyQualifiedVariable, start, count int)
 func getIndexedVariableCount(c *proc.Variable) int {
 	indexedVars := 0
 	switch c.Kind {
-	case reflect.Array, reflect.Slice, reflect.Map:
+	case reflect.Array, reflect.Slice:
 		indexedVars = int(c.Len)
+	case reflect.Map:
+		indexedVars = int(c.Len) * 2
```

Comment:
I think len(c.Children) will give you the single vs double number for array vs map without needing separate cases. However, this logic does not seem right to me. The proc var representing the map always has double the children for the number of map entries (aka key-value pairs). But the dap var we then generate might or might not have the double count. Scalar-indexed map will use the scalar as the name of a child variable and end up with as many children as there are key-value pairs. The number should be double if the key is not a scalar and then we have one child per key and one child per value for each pair.

Reply:
len(Children) is just the ones that are loaded

Reply:
We use this number to load the correct children, and whether we return 2*v.Len or v.Len actual children happens after the request. This may be confusing for the client that they are getting less than the number they requested when they page, but currently they are getting less and it is because there are results missing (whereas this new behavior may have less but none are missing).

If we want to be the most correct about this we would need to check here, before the maybeReloadChildren to determine whether to load 2*count or count, etc. I'll do a quick pass to make that change.

Reply:
Instead of updating the number of children and resolving this case, I instead just made a change to deal with the short circuiting bug in maybeLoadResliced. If we want to accurately represent the number of children in a future change that is an option. Updated the comment for getIndexed to describe what the number represents (since this may not be the same as the actual number of children returned)